src/Api/Controllers/OrganizationsController.cs

```diff
@@ -555,5 +556,62 @@ public async Task PutTaxInfo(string id, [FromBody]OrganizationTaxInfoUpdateReque
             };
             await _paymentService.SaveTaxInfoAsync(organization, taxInfo);
         }
+        
+        [HttpGet("{id}/keys")]
+        public async Task<OrganizationKeysResponseModel> GetKeys(string id)
+        {
+            var user = await _userService.GetUserByPrincipalAsync(User);
+            if (user == null)
+            {
+                throw new UnauthorizedAccessException();
+            }
```

Comment:
Why are we fetching the user?

Reply:
Just to make sure its a valid user making the call. Do you think its not necessary in this context?

Reply:
Discussed. Will update.

Reply:
Updated.